### PR TITLE
adds continuous integration/delivery actions

### PR DESCRIPTION
**Pull request by adds68**
Hi,

I have added 2 Github actions for git-depends.

continuous-integration.yaml - This runs on every push/PR and will build/test the software.

continuous-delivery.yaml - This runs whenever a new release is tagged and publishes the binaries/source code/checksums under the "releases" tab on Github.

I would welcome reviews etc, i hope this helps!

### CODE REVIEW

**Comment by adds68** [comment]
You can view the test jobs on my fork here: https://github.com/adds68/git-depend/actions

File: .github/workflows/continuous-integration.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+
+
+    - name: Build
+      run: go build -v ./...
+
+    - name: Test
+      run: go test -v ./...
```
Comment (finn-ball) [inline .github/workflows/continuous-integration.yaml:21]: No newline at the end of the file.

**Comment by finn-ball** [comment]
Are we able to get this built and test on multiple platforms including Windows and arm??

**Comment by adds68** [comment]
Thanks for the review @finn-ball, i will fix the formatting issues and see if it's possible to enable Github actions on other platforms

**Comment by adds68** [comment]
Reopening on another branch

**Pull request closed without merging**